subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationPublications.java

```diff
@@ -162,6 +172,34 @@ public void artifact(Object notation, Action<? super ConfigurablePublishArtifact
         configureAction.execute(publishArtifact);
     }
 
+    @Override
```

Comment:
This method is nearly identical to `artifacts(Provider, Action)`. I wonder if it's possible for `artifacts(Provider)` to invoke the other, passing an empty, no-op or null Action?

```
public void artifacts(Provider<? extends Iterable<? extends Object>> provider) {
  artifacts(provider, () -> { /* no-op*/ });
}
```